### PR TITLE
[Keyboard] Add Spider Island 78 key ps2avrGB based keyboard

### PR DESCRIPTION
**Pull request by myfreeweb**
Support for this keyboard → https://a.aliexpress.com/_dVJsSpR

Split keyboard with "floating key" design, ps2avrGB stock firmware, non-RGB backlight, mini-USB connection, SATA cable for connecting the halves.

Matrix code copied from `split75` (#8511). Would be nice to reuse it but I don't see any place for "common" matrix code.

## TODO

- [x] JSON
- [ ] better Fn keymap?

### CODE REVIEW

File: keyboards/spiderisland_split/rules.mk [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+WS2812_DRIVER = i2c
+
+# custom matrix setup
+CUSTOM_MATRIX = lite
+SRC = matrix.c
```
Comment (fauxpark) [inline keyboards/spiderisland_split/rules.mk:23]:
```suggestion
SRC += matrix.c
```

File: keyboards/spiderisland_split/rules.mk [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
```
Comment (fauxpark) [inline keyboards/spiderisland_split/rules.mk:17]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
```

File: keyboards/spiderisland_split/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          KC_P0, KC_PDOT, KC_PENT, KC_HOME, KC_PGDN, KC_END
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+#ifdef CONSOLE_ENABLE
+    uprintf("KL: kc: %u, col: %u, row: %u, pressed: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed);
+#endif
+  return true;
+}
```
Comment (fauxpark) [inline keyboards/spiderisland_split/keymaps/default/keymap.c:63]:
```suggestion
```

File: keyboards/spiderisland_split/info.json [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+        },
+        {
+          "x": 77,
+          "y": 0,
+          "label": "r41"
+        }
+      ]
+    }
+  }
+}
```
Comment (fauxpark) [inline keyboards/spiderisland_split/info.json:419]:
```suggestion
{
    "keyboard_name": "Spider Island 78-key split",
    "url": "",
    "maintainer": "qmk",
    "width": 16,
    "height": 6.25,
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"x": 0, "y": 0},
                {"x": 2, "y": 0},
                {"x": 3, "y": 0},
                {"x": 4, "y": 0},
                {"x": 5, "y": 0},
                {"x": 6.5, "y": 0},
                {"x": 8.5, "y": 0},
                {"x": 9.5, "y": 0},
                {"x": 10.5, "y": 0},
                {"x": 12, "y": 0},
                {"x": 13, "y": 0},
                {"x": 14, "y": 0},
                {"x": 15, "y": 0},
                {"x": 0, "y": 1.25},
                {"x": 1, "y": 1.25},
                {"x": 2, "y": 1.25},
                {"x": 3, "y": 1.25},
                {"x": 4, "y": 1.25},
                {"x": 5, "y": 1.25},
                {"x": 6, "y": 1.25},
                {"x": 8, "y": 1.25},
                {"x": 9, "y": 1.25},
                {"x": 10, "y": 1.25},
                {"x": 11, "y": 1.25},
                {"x": 12, "y": 1.25},
                {"x": 13, "y": 1.25},
                {"x": 14, "y": 1.25, "w": 2},
                {"x": 0, "y": 2.25, "w": 1.5},
                {"x": 1.5, "y": 2.25},
                {"x": 2.5, "y": 2.25},
                {"x": 3.5, "y": 2.25},
                {"x": 4.5, "y": 2.25},
                {"x": 5.5, "y": 2.25},
                {"x": 7.5, "y": 2.25},
                {"x": 8.5, "y": 2.25},
                {"x": 9.5, "y": 2.25},
                {"x": 10.5, "y": 2.25},
                {"x": 11.5, "y": 2.25},
                {"x": 12.5, "y": 2.25},
                {"x": 13.5, "y": 2.25},
                {"x": 14.5, "y": 2.25, "w": 1.5},

                {"x": 0, "y": 3.25, "w": 1.75},
                {"x": 1.75, "y": 3.25},
                {"x": 2.75, "y": 3.25},
                {"x": 3.75, "y": 3.25},
                {"x": 4.75, "y": 3.25},
                {"x": 5.75, "y": 3.25},
                {"x": 7.75, "y": 3.25},
                {"x": 8.75, "y": 3.25},
                {"x": 9.75, "y": 3.25},
                {"x": 10.75, "y": 3.25},
                {"x": 11.75, "y": 3.25},
                {"x": 12.75, "y": 3.25},
                {"x": 13.75, "y": 3.25, "w": 2.25},

                {"x": 0, "y": 4.25, "w": 2.25},
                {"x": 2.25, "y": 4.25},
                {"x": 3.25, "y": 4.25},
                {"x": 4.25, "y": 4.25},
                {"x": 5.25, "y": 4.25},
                {"x": 6.25, "y": 4.25},
                {"x": 8.25, "y": 4.25},
                {"x": 9.25, "y": 4.25},
                {"x": 10.25, "y": 4.25},
                {"x": 11.25, "y": 4.25},
                {"x": 12.25, "y": 4.25, "w": 1.75},
                {"x": 14, "y": 4.25},
                {"x": 15, "y": 4.25},

                {"x": 0, "y": 5.25, "w": 1.25},
                {"x": 1.25, "y": 5.25, "w": 1.25},
                {"x": 2.5, "y": 5.25, "w": 1.25},
                {"x": 3.75, "y": 5.25, "w": 1.25},
                {"x": 5, "y": 5.25, "w": 2.25},
                {"x": 8.25, "y": 5.25, "w": 2.25},
                {"x": 10.5, "y": 5.25, "w": 1.25},
                {"x": 11.75, "y": 5.25, "w": 1.25},
                {"x": 13, "y": 5.25},
                {"x": 14, "y": 5.25},
                {"x": 15, "y": 5.25}
            ]
        }
    }
}
```

File: keyboards/spiderisland_split/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        KC_CAPS,    KC_A,    KC_S,    KC_D,    KC_F,    KC_G,
+        KC_LSFT,    KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,
+        KC_LCTL, KC_LGUI, KC_LALT, TT(_FN),  KC_SPC,
+        // right hand
+          KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,  KC_F12,
+           KC_7,    KC_8,    KC_9,    KC_0, KC_MINS,  KC_EQL, KC_BSPC,
+           KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_RBRC, KC_BSLS,
+           KC_H,    KC_J,    KC_K,    KC_L, KC_QUOT, KC_SCLN,  KC_ENT,
+           KC_N,    KC_M, KC_COMM,  KC_DOT, KC_RSFT,   KC_UP, KC_SLSH,
+         KC_SPC, KC_RALT, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```
Comment (fauxpark) [inline keyboards/spiderisland_split/keymaps/default/keymap.c:39]:
```suggestion
        KC_GESC,          KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_QUOT, KC_SCLN,          KC_ENT,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_RSFT, KC_UP,   KC_SLSH,
        KC_LCTL, KC_LGUI, KC_LALT, TT(_FN),          KC_SPC,           KC_SPC,           KC_RALT, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/spiderisland_split/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        BL_TOGG, KC_LEFT, KC_DOWN, KC_RGHT,   KC_NO,   KC_NO,
+        KC_SLCK,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
+          KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
+        // right hand
+         KC_F18,  KC_F19,  KC_F20,  KC_F21,  KC_F22,  KC_F23,  KC_F24,
+          KC_NO,   KC_NO,   KC_NO,   KC_NO,  BL_DEC,  BL_INC,  KC_DEL,
+          KC_P7,   KC_P8,   KC_P9,   KC_NO,   KC_NO,   KC_NO,   KC_NO, KC_NO,
+          KC_P4,   KC_P5,   KC_P6,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
+          KC_P1,   KC_P2,   KC_P3,   KC_NO, KC_SLSH, KC_PGUP, KC_NO,
+          KC_P0, KC_PDOT, KC_PENT, KC_HOME, KC_PGDN, KC_END
```
Comment (fauxpark) [inline keyboards/spiderisland_split/keymaps/default/keymap.c:54]:
```suggestion
        RESET,            KC_F13,  KC_F14,  KC_F15,  KC_F16,  KC_F17,  KC_F18,  KC_F19,  KC_F20,  KC_F21,  KC_F22,  KC_F23,  KC_F24,
        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, BL_DEC,  BL_INC,  KC_DEL,
        KC_NLCK, XXXXXXX, KC_UP,   XXXXXXX, XXXXXXX, XXXXXXX, KC_P7,   KC_P8,   KC_P9,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
        BL_TOGG, KC_LEFT, KC_DOWN, KC_RGHT, XXXXXXX, XXXXXXX, KC_P4,   KC_P5,   KC_P6,   XXXXXXX, XXXXXXX, XXXXXXX,          XXXXXXX,
        KC_SLCK,          XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_P1,   KC_P2,   KC_P3,   XXXXXXX, KC_SLSH, KC_PGUP, XXXXXXX,
        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,          XXXXXXX,          KC_P0,            KC_PDOT, KC_PENT, KC_HOME, KC_PGDN, KC_END
```

File: keyboards/spiderisland_split/spiderisland_split.h [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    l13, l14, l15, l16, l17, l18,  \
+    l19, l20, l21, l22, l23, l24,  \
+    l25, l26, l27, l28, l29, l30,  \
+    l31, l32, l33, l34, l35,  \
+    r00, r01, r02, r03, r04, r05, r06,  \
+    r07, r08, r09, r10, r11, r12, r13,  \
+    r14, r15, r16, r17, r18, r19, r20, r21,  \
+    r22, r23, r24, r25, r26, r27, r28,  \
+    r29, r30, r31, r32, r33, r34, r35,  \
+    r36, r37, r38, r39, r40, r41  \
```
Comment (fauxpark) [inline keyboards/spiderisland_split/spiderisland_split.h:33]:
```suggestion
    l00,      l01, l02, l03, l04, l05, r00, r01, r02, r03, r04, r05, r06, \
    l06, l07, l08, l09, l10, l11, l12, r07, r08, r09, r10, r11, r12, r13, \
    l13, l14, l15, l16, l17, l18, r14, r15, r16, r17, r18, r19, r20, r21, \
    l19, l20, l21, l22, l23, l24, r22, r23, r24, r25, r26, r27,      r28, \
    l25,      l26, l27, l28, l29, l30, r29, r30, r31, r32, r33, r34, r35, \
    l31, l32, l33, l34,      l35,      r36,      r37, r38, r39, r40, r41 \
```

File: keyboards/spiderisland_split/keymaps/default/keymap.c (hunk 5 above)
Comment (myfreeweb) [reply keyboards/spiderisland_split/keymaps/default/keymap.c:39]: I saw the split `// left hand` formatting in multiple split keyboards (`keyboards/ergodone/keymaps/default/keymap.c`, `keyboards/ergodox_infinity/keymaps/default/keymap.c`…), I like it..?

Comment (fauxpark) [reply keyboards/spiderisland_split/keymaps/default/keymap.c:39]: It won't match the ordering of the info.json.

Comment (myfreeweb) [reply keyboards/spiderisland_split/keymaps/default/keymap.c:39]: oh.. well that sucks, I thought info.json would match on the macro argument names :/

File: keyboards/spiderisland_split/config.h [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xFEED
+#define PRODUCT_ID   0x6969
```
Comment (noroadsleft) [inline keyboards/spiderisland_split/config.h:24]:
```suggestion
#define PRODUCT_ID   0xF4E4
```

File: keyboards/spiderisland_split/readme.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# spiderisland_split
+
+[![spiderisland_split](https://i.imgur.com/N3DjdGq.jpg)](https://imgur.com/a/0AUY8Ng)
```
Comment (noroadsleft) [inline keyboards/spiderisland_split/readme.md:3]:
```suggestion
![spiderisland_split](https://i.imgur.com/N3DjdGql.jpg)  
[Image Gallery](https://imgur.com/a/0AUY8Ng)
```

File: keyboards/spiderisland_split/config.h [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xFEED
+#define PRODUCT_ID   0xF4E4
+#define DEVICE_VER   0x0001
+#define MANUFACTURER unrelenting.technology
+#define PRODUCT      spiderisland_split
```
Comment (noroadsleft) [inline keyboards/spiderisland_split/config.h:27]:
```suggestion
#define MANUFACTURER SpiderIsland
#define PRODUCT      Split 78-key
```

File: keyboards/spiderisland_split/readme.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
+# spiderisland_split
```
Comment (noroadsleft) [inline keyboards/spiderisland_split/readme.md:1]:
```suggestion
# SpiderIsland Split 78-key
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Also, It's a bit unusual, but I'm thinking I'd rather have the source folder be `spiderisland/split78` instead of `spiderisland_split`. Who know what other boards might appear from this vendor.

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**